src/main/java/com/icthh/xm/uaa/service/account/password/reset/PasswordResetHandlerFactory.java

```diff
@@ -0,0 +1,35 @@
+package com.icthh.xm.uaa.service.account.password.reset;
+
+import com.icthh.xm.uaa.service.account.password.reset.type.custom.CustomPasswordResetHandler;
+import com.icthh.xm.uaa.service.account.password.reset.type.email.EmailPasswordResetHandler;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.util.Optional.ofNullable;
+
+@Component
+@RequiredArgsConstructor
+public class PasswordResetHandlerFactory {
+    public static final String EMAIL_TYPE = "EMAIL";
+
+    private final CustomPasswordResetHandler customPasswordResetHandler;
+    private final EmailPasswordResetHandler emailPasswordResetHandler;
+
+    private Map<String, PasswordResetHandler> registeredHandlers;
+
+    @PostConstruct
+    private void init() {
+        this.registeredHandlers = new HashMap<>();
+        this.registeredHandlers.put(EMAIL_TYPE, emailPasswordResetHandler);
+    }
+
+    public PasswordResetHandler getPasswordResetHandler(String resetType) {
+        ofNullable(resetType).orElseThrow(() -> new IllegalArgumentException("Reset type must exist"));
+        return ofNullable(registeredHandlers.get(resetType)).orElse(customPasswordResetHandler);
```

Comment:
Map already has `getRrDefault` method for such purposes:
```
 registeredHandlers.getOrDefault(resetType, customPasswordResetHandler)
```

BTW, why do you return by default custom handler instead of email handler that was implemented previously?

Reply:
done

Reply:
sorry, I didn't answer your question 
if we receive some unknown reset type that we don't know about why would we use email handler ?

Reply:
@antonysychov 
if we completely sure that we do not affect old logic then we can use the custom handler. my concern was only regarding backward compatibility.

Reply:
concerning backward compatibility, I'd like to pay your attention (just in case) to the fact that this pr adds new endpoint with the same url but with content-type=application/json.
So, as result, there will be 2 endpoints 
/account/reset_password/init without content-type, accepting string with email
/account/reset_password/init with content-type=application/json, accepting json with {login, loginType, resetType}

basically there shouldn't be any issues with backward compatibility for clients that use api correctly
only clients that send requests with string payload and content-type = application/json (which is incorrect in this case) will be affected since now their requests will be handled by new endpoint that accepts json.

If we don't want to affect current clients that use api incorrectly, probably is't better to add endpoint with different url.
@ishkurko @amedvedchuk what do you think ?

Reply:
if it is really impossible to call the new endpoint with the old request because of content type I think we can leave it as you've done